src/MagnumPlugins/UfbxImporter/UfbxImporter.conf

```
@@ -0,0 +1 @@
+provides=UfbxImporter
```

```suggestion
provides=FbxImporter
```

The plugin system allows requesting a "general" importer, and it will find a matching implementation. Here, you would request `FbxImporter` and it would provide either AssimpImporter or UfbxImporter, depending on what's available.

---

Ah that makes sense, one thing to think about is whether we want to enable the `.obj` format support that ufbx has (well, when I merge the branch, but it's getting very solid now, ie. heavily fuzzed, enough real world test cases, etc). I checked that Magnum currently has two .obj importers: `ObjImporter` with "severe" caveats and the one contained in `AssimpImporter`.

I implemented .obj support for *ufbx* as it fits the internal data model very well and is a tiny amount of code in comparison to anything in FBX. It should support some more features than Assimp (eg. MRGB vertex colors) and in general be up to the same standard of testing/support as *ufbx* proper.

So if the .obj file format is enabled should probably advertise both here as:

```
provides=FbxImporter
provides=ObjImporter
```

---

Haha, I was checking your repo today, saw the OBJ support branch and thought "mmm nice" :)

The builtin ObjImporter is (over)due for a rewrite for a similar reason, to remove the need for Assimp in yet another use case. I have a private branch with some changes, but it'll take quite some time to get finished, as I want to abuse it as a playground for various (text) parsing optimizations.

So yes -- adding `provides=ObjImporter` makes sense. The more alternatives, the better.

---

Nice, I'll add that in then!

For text parsing optimizations you might find my small-but-quite-efficient (at least on x64 with hardware 128-bit division) [double parsing](https://github.com/bqqbarbhg/ufbx/blob/67f756329f20dbbea50205b16bf80dcdbe2ff908/ufbx.c?ts=4#L868-L1115) interesting. It's not as fast as [Lemire's](https://github.com/lemire/fast_double_parser) and it has to defer to `strtod()` in "hard" cases but I think it has a neat complexity/speed tradeoff.

---

Thanks!

For a floating-point parsing replacement I was considering a rather hardcore goal for web builds -- to not have to fall back to `strtod()` (so it doesn't need to drag all its complexity inside the webassembly binary), and to have a possibility to opt out of double-precision parsing at compile time, leaving just 32bit floats (so e.g. if there are any large LUTs for double-precision parsing, those would not get included).

I have to check again if that has any practical usefulness, though -- if one needs a as-tiny-as-possible web build, then string parsing is probably out of question anyway :D